### ISSUE TITLE
Really slow attacks with demo code

### ISSUE THREAD

**Issue by renebidart** [bethgelab/AnalysisBySynthesis#1]
I'm trying to run the demo in attacks.ipynb, but it is really slow, taking over 30 minutes to run the [PointwiseAttack](https://github.com/bethgelab/AnalysisBySynthesis/blob/458ee76af68778fd9a0ea7c4b411fa4c7eeb8682/scripts/attacks.py#L73).

 I'm using GPU and running the  [classification](https://github.com/bethgelab/AnalysisBySynthesis/blob/458ee76af68778fd9a0ea7c4b411fa4c7eeb8682/scripts/attacks.py#L66) was quite fast, but I'm wondering if I may be doing something wrong, or is it normally like this?

Thanks

**Comment by wielandbrendel** [comment]
@lukas-schott Could you take a look?

**Comment by jaks19** [comment]
I am not using GPU but I am also thinking something might be wrong on my end if after 20min I still did not get the adversarial example.

**Comment by lukas-schott** [comment]
Yes, the pointwise attack is quite slow ~30 min / adversarial on a Titan X. 

I implemented some speedups: switched to an attack which requires less iterations and changed the default model to an approximation of the ABS model (n_iter=10 instead of n_iter=50). It should now take ~1-5 min to create an adversarials on a GPU with the provided code.
